### DISCUSSION TITLE
Text classification in different languages

### DISCUSSION THREAD

**Discussion by the42** [ludwig-ai/ludwig#3250] [General]
I followed the helpful tutorial on [Text classification](https://ludwig.ai/latest/examples/text_classification/). It seems this type of classification assumes English input? What would the yaml configuration look like if my target language is german?

What languages does ludwig support in general?

**Comment by arnavgarg1** [comment]
Hi @the42, the tutorial does assume the text input is in English, however, this is not a requirement!

Ludwig does support at least one text encoder out of the box that is multilingual: [XLMRoberta](https://ludwig.ai/latest/configuration/features/text_features/#xlmroberta). Here's a masking example I used that's in [Hindi](https://huggingface.co/jplu/tf-xlm-roberta-base?text=%E0%A4%B8%E0%A5%8D%E0%A4%AA%E0%A5%87%E0%A4%A8+%3Cmask%3E+%E0%A4%AE%E0%A5%87%E0%A4%82+%E0%A4%8F%E0%A4%95+%E0%A4%B6%E0%A4%B9%E0%A4%B0+%E0%A4%B9%E0%A5%88). I also tried an equivalent in Spanish.

Additionally, if you would like, you can use a pretrained multilingual text encoder from Huggingface by using the [AutoTransformer text encoder](https://ludwig.ai/latest/configuration/features/text_features/#autotransformer) and specifying the pretrained model/path from Huggingface.

Let me know if this helps!
